torch/_decomp/decompositions.py

```diff
@@ -3354,7 +3354,8 @@ def get_values(inp, out, scales, nsqueeze):
         i = torch.arange(out, device=input.device).to(dtype=dtype)
 
         x_f32 = _compute_source_index(scale_factor, i, align_corners).clamp(min=0.0)
-        x_f32 = x_f32.reshape(len(x_f32), *[1] * (nsqueeze))
+        for _ in range(nsqueeze):
+            x_f32 = x_f32.unsqueeze(-1)
```

Comment:
oh my. Could you please open an issue with this weird behaviour?

Reply:
It was my  mistake. I called `len` which has to give a int object and torch.compile was generating code specialized to that length, which is the correct behaviour from the compiler. Note to self: don't call `len` on tensors in decomps. (Or any method that outputs a builtin type instead of a symbolic type)